### PR TITLE
feat: add Tags Input feature

### LINKED ISSUE #8766: [FEATURE] enhance tag input with TagsInput component
### Description

I would like to suggest enhancing the current input tag field in the [pages/account/manage/profile.js](https://github.com/EddieHubCommunity/LinkFree/blob/e23cd822a9d931bc2174fa1464bbca7be5855378/pages/account/manage/profile.js#L223C1-L228C25) by implementing the **`TagsInput`** component. Currently the input tag allows user to add tags using a simple input field. However, this doesn't provide a structured way to categorize or organize tags, by implementing a **`TagsInput`** component we can provide more intuitive and user-friendly experience for managing tags.

**Proposed implementation**
* The **`TagsInput`** component will consist of an input field where users can type tags, when the input field is focused and the user presses(**`,`**) key, if the input field is not empty, tags will be added and displayed  as individual elements
* When the input field is focused and the user presses(**`BACKSPACE`**) key, if the input field is empty and there are tags present, the last-added tag will be removed and displayed again in the input field.
* Users can remove a tag by clicking on the (**`X`**) button associated with each tag

### Screenshots

Manage Profile - Google Chrome 2023-08-24 21-52-27

### Additional information

I'm excited to work on this feature if the idea sounds good.
Looking forward to your feedback and guidance on how to proceed with this feature suggestion. Thank you for considering it

### PR DESCRIPTION
**Pull request by badrivlog**
## Fixes Issue

Closes #8766 
## Changes proposed
* Created a new **`TagsInput`** component for inputting and displaying tags
* Implement a  **`handleKeyUp`** event handler for separate tags, when the **`COMMA`** key presses. If the input field is not empty, tags will be added and displayed as individual elements.
* Implement a **`handleKeyDown`** event handler for the **`BACKSPACE`** key when the input field is focused. Check if the input field is empty and the user presses **`BACKSPACE`**. If the conditions are met, remove the last tag from the state and update the input field to display the removed tag
* Attach an **`onClick`** event handler to each rendered tag element, when a tag is clicked, call a function to remove the clicked tag from the state.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
TagsInput

dark mode:
tagsInput dark

### CODE REVIEW

File: components/tag/TagsInput.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      className="flex items-center gap-x-1 text-sm p-1 font-mono border rounded-md line-clamp-1"
+    >
+      <span>{tag}</span>
+      <button
+        type="button"
+        className="inline-block text-center"
+        onClick={() => onTagRemove(tag)}
+      >
+        <span className="absolute h-px w-px whitespace-nowrap overflow-hidden">
+          remove {tag} tag
```
Comment (badrivlog) [inline components/tag/TagsInput.js:45]: These words are not needed by sighted users, they are there to provide more information about what the button does for screen reader users who do not have the extra visual context to help them.

File: components/tag/TagsInput.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        </span>
+        <span aria-hidden="true">
+          <XMarkIcon
+            className="w-4 h-4 hover:text-tertiary-medium"
+            title={`Remove ${tag}`}
+          />
+        </span>
```
Comment (badrivlog) [inline components/tag/TagsInput.js:52]: This icon would be read out in screen readers ("cross mark"). Using `aria-hidden` puts a stop to that

File: components/tag/TagsInput.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  const keyCode = {
+    comma: 188,
+    backspace: 8,
+  };
+
+  const handleKeyUp = (e) => {
+    const inputValue = inputRef.current.value;
+    const { comma } = keyCode;
+    if (e.keyCode !== comma) return;
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:16]: We don't use this code styling anywhere, so I think better to use brackets

```suggestion
    if (e.keyCode !== comma) {
       return;
    }
```

File: components/tag/TagsInput.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    const newTag = inputValue.trim().replace(/,/g, "");
+    if (!newTag) return;
```
Comment (eddiejaoude) [inline components/tag/TagsInput.js:18]: Same here

File: pages/account/manage/profile.js [hunk 5]
```diff
@@ -83,6 +84,16 @@ export default function Profile({ BASE_URL, profile, fileExists }) {
     };
   });
 
+  const handleTagAdd = (newTag) => {
+    setTags((prevState) => [...prevState, newTag]);
+  };
+
+  const handleTagRemove = (tagToRemove) => {
+    const indexOfTagToRemove = tags.findIndex((tag) => tag === tagToRemove);
+    const updatedTags = tags.filter((_, index) => index !== indexOfTagToRemove);
```
Comment (eddiejaoude) [inline pages/account/manage/profile.js:93]: Is `findIndex` needed? I think this can be done with just a `filter`?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good and works great 👍

**Comment by badrivlog** [comment]
> Looks good and works great 👍 

Thanks for reviewing the PR🎉 I will make changes as you suggested

**Comment by eddiejaoude** [comment]
Thank you @badrivlog , sorry for being slow in reviewing, I was away at events all last week and weekend, so catching up now

Awesome! Then we can merge 🎉

**Review by eddiejaoude — APPROVED** [review]
Looks good, thank you 👍

**Review by xkrishguptaa — APPROVED** [review]

**Pull request merged**